### PR TITLE
Add expire functionality

### PR DESCRIPTION
**Pull request by craff**
Solving issue #7

### CODE REVIEW

**Comment by craff** [comment]
Remark: when doing the commit I saw a lot of changes in the documentation that I did not commit.

File: lib/http_cookie.ml [hunk 1]
```diff
@@ -797,3 +797,13 @@ let update_same_site same_site cookie = { cookie with same_site }
 let update_extension extension cookie =
   let+ extension = parse_opt extension_value extension in
   { cookie with extension }
+
+let delete cookie =
```
Comment (bikallem) [inline lib/http_cookie.ml]: Could you name the function `expire`. It is more inline with what we are doing.

File: lib/http_cookie.mli [hunk 2]
```diff
@@ -223,3 +223,7 @@ val update_secure : bool -> t -> t
 val update_http_only : bool -> t -> t
 val update_same_site : same_site option -> t -> t
 val update_extension : string option -> t -> (t, string) result
+
+(** mark the Cookie with a negative max-age, remove now useless payload.
```
Comment (bikallem) [inline lib/http_cookie.mli]: Please follow the documentation format as followed by other functions. Could you run `ocamlformat` on the changes.

**Comment by bikallem** [comment]
The PR looks good module some requested changes. Also required are unit tests.

**Comment by craff** [comment]
I added one test ... and found a bug ;-) This is fixed. I will put one or two extra test.

**Comment by craff** [comment]
I do not see your requested changes ?

**Comment by craff** [comment]
The parser is a bit lax now. It accepts 01 or -01. Do we want to be more strict to warrant
that printing o parsing = idt ?

**Comment by bikallem** [comment]
> I do not see your requested changes ?

I meant the review comments:
1) Could you name the function expire. It is more inline with what we are doing.
2.) Please follow the documentation format as followed by other functions. Could you run ocamlformat on the changes.

**Comment by bikallem** [comment]
> The parser is a bit lax now. It accepts 01 or -01. Do we want to be more strict to warrant
> that printing o parsing = idt ?

I think the parser as it is is okay since 0 is a valid Max-Age value.

**Comment by craff** [comment]
I do not know where to see the "review comment". This is the second time this happens to me on github ;-)
Thanks for putting you comments here.

 I did the above change (expire + doc), except ocamlformat which gives an error, and I do not want to do something silly:
```
ocamlformat http_cookie.ml
ocamlformat: Error while parsing /home/raffalli/Programming/http-cookie/.ocamlformat:
             Project should be formatted using ocamlformat version "0.20.1", but the installed version is "0.24.1"
```

**Comment by craff** [comment]
Moreover, I just installed mdx so that dune runtest does not complain and I got this:
```
File "README.md", line 1, characters 0-0:
diff --git a/_build/default/README.md b/_build/default/.mdx/README.md.corrected
index eb92e83..6aa709d 100644
--- a/_build/default/README.md
+++ b/_build/default/.mdx/README.md.corrected
@@ -16,6 +16,7 @@ The RFC standards implemented by the library are:
 '''ocaml
 # #require "http-cookie";;
 # #install_printer Http_cookie.pp;;
+Cannot find type Topdirs.printer_type_new.
 '''
 
 HTTP `Cookie` header value:
@@ -42,16 +43,5 @@ Create HTTP cookie:
 
 '''ocaml
 # Http_cookie.create ~path:"/home" ~domain:"eee:aaa:abdf::223:192.168.0.1" ~secure:true ~same_site:`Strict ~name:"SID" "31d4d96e407aad42";;
-- : (Http_cookie.t, string) result =
-Ok
- name: SID
- value: 31d4d96e407aad42
- path: /home
- domain: eee:aaa:abdf::223:192.168.0.1
- expires:
- max_age:
- secure: true
- http_only: true
- same_site: Strict
- extension:
+- : (Http_cookie.t, string) result = Ok <abstr>
 '''
```

None of this seems related to this PR?

**Comment by craff** [comment]
I also opened an issue about the documentation. But I can not do the PR since it requires creating a new branch
and changing the way you put the documentation online probably.

**Comment by bikallem** [comment]
> Moreover, I just installed mdx so that dune runtest does not complain and I got this

What is the version of mdx that you have in your opam switch? Looks like mdx is not working correctly now. Regardless, this issue is not related to this PR, so you can do `dune runtest lib` for now.

**Comment by bikallem** [comment]
> I did the above change (expire + doc), except ocamlformat which gives an error, and I do not want to do something silly:

~~I think you have to install the version of ocamlformat as suggested by the error message.~~ I have updated the ocamlformat version to 0.24.1 (the latest which also matches the version you are using). 

I tried to push to your PR branch but couldn't since it was not allowed. Therefore please rebase on master.

**Comment by craff** [comment]
done ocamlformat

**Comment by craff** [comment]
mdx 2.1.0 with ocaml 4.13.1 reports no error
mdx 2.1.0 with ocaml 5.0.0 reports the above error (on master branch too, so I do a sperate issue)

**Comment by bikallem** [comment]
> mdx 2.1.0 with ocaml 5.0.0 reports the above error (on master branch too, so I do a sperate issue)

Looks like the cause is either a mdx or ocaml 5.0. So, yes this is a separate issue.

**Review by bikallem — APPROVED** [review]

**Comment by bikallem** [comment]
Thanks.

**Pull request merged**